scripts/dudle.coffee

```diff
@@ -99,82 +100,70 @@ has_trailing_slash = (str) ->
     pattern.test str
 
 is_already_subscribed = (shortname) ->
-    dudles = read_dudles_file()
-    found = false
-    for dudle in dudles
-        if dudle.shortname == shortname
-            found = true
-            break
-    found
+    read_dudles_file().has shortname
 
 read_dudles_file = ->
-    dudles = []
     try
-        dudles = JSON.parse(fs.readFileSync(dudle_db))
+        dudles = new Map(JSON.parse(fs.readFileSync(dudle_db)).map((dudle) -> [dudle.shortname, dudle]))
     catch err
+        dudles = new Map()
         console.log "Couldn't find #{dudle_db}"
     dudles
 
-write_dudles_file = (dudle_list) ->
+write_dudles_file = (dudle_map) ->
     try
-        fs.writeFile(dudle_db, JSON.stringify(dudle_list, null, 2))
+        fs.writeFile(dudle_db, JSON.stringify(Array.from(dudle_map.values()), null, 2))
     catch err
         console.log "Couldn't write to #{dudle_db}: #{err}"
 
 save_dudle_to_file = (shortname, url) ->
     dudles = read_dudles_file()
     new_dudle =
-        "shortname": shortname,
-        "url": url,
-        "last_checked": new Date().toISOString()
-    dudles.push new_dudle
+        shortname: shortname
+        url: url
+        last_checked: new Date().toISOString()
+    dudles.set shortname, new_dudle
     write_dudles_file dudles
 
 remove_dudle_from_file = (shortname) ->
     dudles = read_dudles_file()
-    new_dudle_list = []
-    found = false
-    for dudle in dudles
-        if shortname != dudle.shortname
-            new_dudle_list.push dudle
-        else
-            found = true
-    write_dudles_file new_dudle_list
-    found
+
+    if dudles.has shortname
+        dudles.delete shortname
```

Comment:
Yes. The power of the new JavaScript `Map`. Thats why I asked if it understands ECMA 6, because as far as I know this was only introduced in 6.